File: dotnet/Samples/LanguageDetection/File/FileLanguageDetectionService.cs
using LanguageWeaver.Sdk.Common;
using LanguageWeaver.Sdk.Configurations;
using LanguageWeaver.Sdk.Constants;
using LanguageWeaver.Sdk.LanguageDetection.Common.Request;

namespace LanguageWeaver.Sdk.Samples.LanguageDetection.File;

public class FileLanguageDetectionService
{
    public static void Main()
    {
        using var lwClient = new SdkFactory().GetLanguageWeaverClient(new ClientConfiguration());
        var fileLanguageDetectionRequest = new FileLanguageDetectionRequest()
        {
            InputFile = Path.Combine(Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName,
                "Resources", "Input", "input1.txt"),
            InputFormat = Format.Plain
        };

        var languageDetectionResult = lwClient.DetectLanguage(fileLanguageDetectionRequest);

        languageDetectionResult.Languages
            .ForEach(language => Console.WriteLine(language.Name));
    }
}

File: java/src/main/java/com/languageweaver/sdk/samples/languagedetection/file/FileLanguageDetectionService.java
package com.languageweaver.sdk.samples.languagedetection.file;

import com.languageweaver.sdk.common.LanguageWeaverClient;
import com.languageweaver.sdk.common.SdkFactory;
import com.languageweaver.sdk.common.configurations.ClientConfiguration;
import com.languageweaver.sdk.common.constants.Format;
import com.languageweaver.sdk.languagedetection.common.request.FileLanguageDetectionRequest;
import com.languageweaver.sdk.languagedetection.common.result.LanguageDetectionResult;

import java.nio.file.Paths;

public class FileLanguageDetectionService {
    public static void main(String[] args) throws Exception {
        try (LanguageWeaverClient lwClient = new SdkFactory().getLanguageWeaverClient(new ClientConfiguration())) {
            FileLanguageDetectionRequest fileLanguageDetectionRequest = new FileLanguageDetectionRequest()
                    .setInputFile(Paths.get("java", "src", "main", "resources", "input", "input1.txt").toFile().getAbsolutePath())
                    .setInputFormat(Format.PLAIN);

            final LanguageDetectionResult languageDetectionResult = lwClient.detectLanguage(fileLanguageDetectionRequest);

            if (languageDetectionResult.getLanguages() != null) {
                languageDetectionResult.getLanguages()
                        .forEach(language -> System.out.println(language.getName()));
            }
        }
    }
}
